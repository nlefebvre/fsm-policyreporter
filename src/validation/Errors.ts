export enum FSMPart {
  STATE = "State",
  ALPHABET = "Alphabet",
  TRANSITION_FN = "Transition function"
}

export interface ErrorMsg {
  inputValue: string | Array<string> | null;
  type: FSMPart;
  message: string;
}

export const generateErrorMessage = (title: string, errors: Array<ErrorMsg>) => {
  return errors.reduce<string>((errorString, { type, message, inputValue }) => {
    return errorString += `    ${type}: ${message} ${inputValue !== null ? `(For value '${inputValue}')` : ''}\n`
  }, title + "\n")
}