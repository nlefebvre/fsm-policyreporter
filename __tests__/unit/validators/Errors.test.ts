
import { FSMPart, generateErrorMessage } from "../../../src/validation/Errors";

describe("generateErrorMessage Tests", () => {
  test("generates error for empty string", () => {
    expect(generateErrorMessage("", [])).toMatchInlineSnapshot(`
"
"
`)
  });


  test("generates message for single error", () => {
    expect(generateErrorMessage("Test error", [{
      type: FSMPart.ALPHABET,
      message: "test message",
      inputValue: "value"
    }])).toMatchSnapshot();
  });

  test("generates message for multiple errors", () => {
    expect(generateErrorMessage("Test error", [
      {
        type: FSMPart.ALPHABET,
        message: "test alphabet message",
        inputValue: null
      },
      {
        type: FSMPart.STATE,
        message: "test state message",
        inputValue: null
      },
      {
        type: FSMPart.TRANSITION_FN,
        message: "test tnfn message",
        inputValue: "value"
      },
    ])).toMatchSnapshot();
  });

});