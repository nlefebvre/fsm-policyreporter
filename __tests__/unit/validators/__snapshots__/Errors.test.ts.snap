// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateErrorMessage Tests generates message for multiple errors 1`] = `
"Test error
    Alphabet: test alphabet message 
    State: test state message 
    Transition function: test tnfn message (For value 'value')
"
`;

exports[`generateErrorMessage Tests generates message for single error 1`] = `
"Test error
    Alphabet: test message (For value 'value')
"
`;
